test/integration/runner_task_test.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+require 'kubernetes-deploy/runner_task'
+class RunnerTaskTest < KubernetesDeploy::IntegrationTest
+  def test_works
+    deploy_fixtures("hello-cloud")
+
+    task_runner = KubernetesDeploy::RunnerTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+
+    task_runner.run(
+      task_template: 'hello-cloud-template-runner',
+      command: nil, # idk
+      args: ['a']
+    )
+
+    assert_logs_match(/Starting task runner/)
+  end
+
+  def test_substitutes_arguments
+    deploy_fixtures("hello-cloud")
+
+    task_runner = KubernetesDeploy::RunnerTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+
+    pod = task_runner.run(
+      task_template: 'hello-cloud-template-runner',
+      command: nil, # idk
+      args: ['a']
+    )
+
+    assert_equal ['a'], pod.spec.containers.first.args
+  end
+
+  def test_missing_namespace
+    assert_raises(KubernetesDeploy::RunnerTask::FatalTaskRunError) do
+      task_runner = KubernetesDeploy::RunnerTask.new(
+        context: KubeclientHelper::MINIKUBE_CONTEXT,
+        namespace: "missing",
+      )
+
+      task_runner.run(
+        task_template: 'hello-cloud-template-runner',
+        command: nil, # idk
+        args: 'a'
+      )
+    end
+  end
+
+  def test_missing_arguments
```

Comment:
These last two should be unit tests. Since they don't rely on a live namespace, let's avoid the overhead of creating/destroying one (which happens for every integration test case).